atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -8,315 +8,66 @@
-                  '--author={name} <{email}>'.format(name=self.author_name,
-                                                     email=self.author_email),
-                  '--message={message}'.format(message=self.commit_message)])
-
-        # Push it
-        self.log.info("Pushing to git repository")
-        repo.git(['push', remote])
-
-    def verify_branch(self, branch, branch_sha):
-        """
-        Raise exception if the branch is not at the correct commit.
-
-        :param branch: str, branch name
-        :param branch_sha: str, commit hash expected
-        """
-
-        if branch_sha != self.git_ref:
-            self.log.error("Branch '%s' is at commit %s (expected %s)",
-                           branch, branch_sha, self.git_ref)
-            raise RuntimeError("Not at expected commit")
-
-        self.log.info("Branch '%s' is at expected commit (%s)",
-                      branch, self.git_ref)
+        self.target = target
+        self.xmlrpc = create_koji_session(hub)
 
     def run(self):
         """
         run the plugin
         """
 
-        if self.workflow.build_process_failed:
-            self.log.info("Build already failed, not incrementing release")
+        parser = DockerfileParser(self.workflow.builder.df_path)
+        release_labels = get_all_label_keys('release')
+        dockerfile_labels = parser.labels
+        if any(release_label in dockerfile_labels
+               for release_label in release_labels):
+            self.log.debug("release set explicitly so not incrementing")
             return
 
-        # Ensure we can use the git repository already checked out for us
-        source = self.workflow.source
-        assert isinstance(source, GitSource)
-        with GitRepo(source.get(), self.log) as repo:
-            # Note: when this plugin is configured by osbs-client,
-            # source.git_commit (the Build's source git ref) comes from
-            # --git-branch not --git-commit. The value from --git-commit
-            # went into our self.git_ref.
-            branch = source.git_commit
-            try:
-                branch_sha = repo.git(['rev-parse', branch])
-            except subprocess.CalledProcessError:
-                self.log.error("Branch '%s' not found in git repo",
-                               source.git_commit)
-                raise RuntimeError("Branch '%s' not found" % branch)
+        # No release labels are set so set them.
+        component_label = get_preferred_label_key(dockerfile_labels,
+                                                  'com.redhat.component')
+        try:
+            component = dockerfile_labels[component_label]
+        except KeyError:
+            raise RuntimeError("missing label: {}".format(component_label))
 
-            # We checked out the right branch
-            assert repo.git(['rev-parse', 'HEAD']) == branch_sha
+        latest = self.xmlrpc.getLatestBuilds(self.target, package=component)
```

Comment:
See https://github.com/projectatomic/atomic-reactor/pull/516 for fix.
Testing was successful.